lib/ansible/modules/cloud/azure/azure_rm_image.py

```diff
@@ -0,0 +1,363 @@
+                "caching": "ReadOnly",
+                "osType": "Linux"
+                },
+                "dataDisks": [
+                    {
+                        "caching": "ReadOnly",
+                        "diskSizeGB": 64,
+                        "storageAccountType": "Standard_LRS",
+                        "managedDisk": {
+                        "id": "/subscriptions/XXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXX/resourceGroups/Testing/providers/Microsoft.Compute/disks/testvm001-datadisk-0"
+                        },
+                        "lun": 0
+                    }
+                ]
+            },
+            "sourceVirtualMachine": {
+                "id": "/subscriptions/XXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXX/resourceGroups/Testing/providers/Microsoft.Compute/virtualMachines/testvm001"
+            }
+          }
+'''  # NOQA
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase, format_resource_id
+
+try:
+    from msrestazure.tools import parse_resource_id
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.compute.models import SubResource, OperatingSystemStateTypes, ImageStorageProfile, \
+                                          Image, ImageOSDisk, ImageDataDisk
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+class AzureRMImage(AzureRMModuleBase):
+
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            resource_group=dict(type='str', required=True),
+            name=dict(type='str', required=True),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            location=dict(type='str'),
+            source=dict(type='str', required=True),
+            data_disk_sources=dict(type='list', default=[]),
+            os_type=dict(type='str', choices=['Windows', 'Linux'])
+        )
+
+        self.results = dict(
+            changed=False,
+            state=dict()
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.state = None
+        self.location = None
+        self.source = None
+        self.data_disk_sources = None
+        self.os_type = None
+        self.tags = None
+
+        super(AzureRMImage, self).__init__(self.module_arg_spec, supports_check_mode=True)
+
+    def exec_module(self, **kwargs):
+
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+
+        results = dict()
+        changed = False
+        image = None
+
+        if not self.location:
+            # Set default location
+            resource_group = self.get_resource_group(self.resource_group)
+            self.location = resource_group.location
+
+        self.log('Fetching image {0}'.format(self.name))
+        image = self.get_image()
+        if image:
+            self.check_provisioning_state(image, self.state)
+            results = self.image_to_dict(image)
+            # update is not supported

```

Comment:
What's the typical E2E scenario for image management? For ACI there is no update API either. But is it possible that user want to update the image to reflect the latest state of the source VM/Blob?